### ISSUE TITLE
Now www.leetcode.com is protected by cloudflare

### ISSUE THREAD

**Issue by tan-wei** [tan-wei/leetcode-rust#1]
So we should bypass cloudflare to success to get problems.

**Comment by tan-wei** [comment]
See [here](https://github.com/kawre/leetcode.nvim/blob/6ac0f12143cc8adbc83fc055e4e73861ac78030a/lua/leetcode/api/headers.lua) to solve this.

**Comment by tan-wei** [comment]
Cookie is required now, which makes the project hard to use. But I'll try.

**Comment by tan-wei** [comment]
Finished by #2 .

**Comment by tan-wei** [comment]
Seems now it does not work since today ...

**Comment by xymeng16** [comment]
> Seems now it does not work since today ...

Cloudflare seems not the reason making this repo unusable. `api/problems/algorithms/` is now using brotli as its compression encoding method and this repo only enabled `gzip` by default.

Two solutions (do some modifications in `src/fetcher.rs:get_problems()`):
1. Enabling all encodings mentioned in the `Accept-Encoding` header field by
```rust
let client = reqwest::blocking::Client::builder()
        .connection_verbose(true)
        .http2_prior_knowledge()
        .gzip(true)
        .deflate(true)
        .zstd(true)
        .brotli(true)
        .build()
        .unwrap();
```
and adding the corresponding features in `Cargo.toml`.
2. Unset `Accept-Encoding` in the header by commenting
```rust
h.insert(
            "Accept-Encoding",
            reqwest::header::HeaderValue::from_static("gzip, deflate, br, zstd"),
        );
```

I have tested both worked.

**Comment by tan-wei** [comment]
@xymeng16 Great. I'll try ASAP.

**Comment by tan-wei** [comment]
Closed by [8d51cdd](https://github.com/tan-wei/leetcode-rust/commit/8d51cdd5ad5c9a6925da2a4220148536181dd089), thanks @xymeng16 .
